File: mahaam-api-py/feat/user/user_model.py
from uuid import UUID
from datetime import datetime
from dataclasses import dataclass
from pydantic import BaseModel, Field


@dataclass
class User:
    id: UUID
    email: str | None = None
    status: str | None = None
    name: str | None = None


class User2(BaseModel):
    id: UUID
    email: str | None = None
    status: str | None = None
    name: str | None = None
    user_id: UUID | None = Field(default=None, serialization_alias="userId")


class Device(BaseModel):
    id: UUID | None = None
    user_id: UUID | None = Field(default=None, serialization_alias="userId")
    platform: str | None = None
    fingerprint: str = None
    info: str | None = None
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")


class SuggestedEmail(BaseModel):
    id: UUID
    user_id: UUID = Field(serialization_alias="userId")
    email: str | None = None
    created_at: datetime | None = Field(default=None, serialization_alias="createdAt")


class VerifiedUser(BaseModel):
    user_id: UUID = Field(serialization_alias="userId")
    device_id: UUID = Field(serialization_alias="deviceId")
    jwt: str
    user_full_name: str | None = Field(default=None, serialization_alias="userFullName")
    email: str | None = None


class CreatedUser(BaseModel):
    id: UUID
    device_id: UUID = Field(serialization_alias="deviceId")
    jwt: str

File: mahaam-api-go/feat/models/user_model.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type User struct {
	ID    uuid.UUID `json:"id,omitempty"`
	Email *string   `json:"email,omitempty"`
	Name  *string   `json:"name,omitempty"`
}

type User2 struct {
	User
	UserID *uuid.UUID `json:"userId"`
}

type Device struct {
	ID          uuid.UUID `json:"id" db:"id"`
	UserID      uuid.UUID `json:"userId" db:"user_id"`
	Platform    string    `json:"platform" db:"platform"`
	Fingerprint string    `json:"fingerprint" db:"fingerprint"`
	Info        string    `json:"info" db:"info"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
}

type SuggestedEmail struct {
	ID        uuid.UUID  `json:"id"`
	UserID    uuid.UUID  `json:"userId"`
	Email     *string    `json:"email"`
	CreatedAt *time.Time `json:"created_at"`
}

type VerifiedUser struct {
	UserID       uuid.UUID `json:"userId"`
	DeviceID     uuid.UUID `json:"deviceId"`
	Jwt          string    `json:"jwt"`
	UserFullName *string   `json:"userFullName"`
	Email        *string   `json:"email"`
}

type CreatedUser struct {
	ID       uuid.UUID `json:"id"`
	DeviceID uuid.UUID `json:"deviceId"`
	Jwt      string    `json:"jwt"`
}

type Meta struct {
	UserID   uuid.UUID
	DeviceID uuid.UUID
}
